src/shared.c

```diff
@@ -1026,6 +1026,14 @@ _Bool ValidateUserInfo(const char *s) {
 void SetUserInfo(char *s, const char *key, const char *value) {
 	char newi[MAX_USER_INFO_STRING], *v;
 
+	if (!value) {
+		return;
+	}
+
+	if (*value == '\0') {
+		return;
+	}
+
```

Comment:
I see how passing null returns the function early (just after del user info) so the intended use `SetUserInfo(s, "active", NULL);` to delete user info is met nicely.

The thing is when `value = NULL` doing a `*value` will segfault. In expression `!value || *value == '\0'` one might tought the same thing might happen. If it doesn't happen I think its because compiler is smart enough to see it and not evaluate right hand side of the || when `value == NULL`